torch/_inductor/fx_passes/post_grad.py

```diff
@@ -216,6 +217,54 @@ def is_valid_mm_plus_mm(match: Match):
     return True
 
 
+def scatter_upon_allzero_extra_check(m):
+    allzero_shape = m.kwargs["shape"]
+    selector = m.kwargs["selector"]
+    val = m.kwargs["val"]
+
+    # Only match scatter against 2D tensors for now.
+    if len(allzero_shape) != 2:
+        return False
+    if not isinstance(val, float):
+        return False
+
+    selector_ft = selector.meta["val"]
+    if selector_ft.dim() == 2 and selector_ft.size(1) != 1:
+        return False
+    return True
+
+
+@register_lowering_pattern(
+    CallFunction(
+        aten.scatter,
+        CallFunction(aten.full, KeywordArg("shape"), 0.0, dtype=KeywordArg("dtype")),
```

Comment:
I think we can support an arbitrary `val` here, right?

Reply:
Below you just need to rewrite it to 

```
        return ops.where(selector == ops.index_expr(idx[1], torch.int64), val, matched_constant)
```

Reply:
Good point. There is no harm to make it more general. Will do that.

I'll rename the patter from `scatter_upon_allzero` to `scatter_upon_const_tensor` then

Reply:
The 2 op pattern match - optimal :^)